src/compiler/checker.ts

```diff
@@ -24575,10 +24575,11 @@ namespace ts {
         }
 
         /**
-         * Returns true iff the JSX element name would be a valid JS identifier, ignoring restrictions about keywords not being identifiers
+         * Returns true if the hyphen character is not in the JSX element name. The hyphen character, if present,
+         * would prevent it from being a valid JS identifier, ignoring restrictions about keywords not being identifiers
          */
         function isUnhyphenatedJsxName(name: string | __String) {
-            // - is the only character supported in JSX attribute names that isn't valid in JavaScript identifiers
+            // - is one of only two characters supported in JSX attribute names that isn't valid in JavaScript identifiers, : being the other

```

Comment:
From my quick glance at this function’s usage, it seemed to me like the note about identifier validity isn’t actually relevant. On second thought, I would probably just remove these comments since they’re subtly misleading about the purpose of the function. The name and one-line implementation of the function are sufficiently self-explanatory.